ext/storage_sources/local_store/local_store.c

```diff
@@ -0,0 +1,1220 @@
+{
+    LOCAL_STORAGE *local;
+    LOCAL_FLUSH_ITEM *flush, *safe_flush;
+    int ret, t_ret;
+    char *match;
+
+    (void)config; /* Unused */
+
+    /*
+     * This implementation does not do anything meaningful on flush. However, we do track which
+     * objects have not yet been flushed and note which ones need to be flushed now.
+     */
+    ret = 0;
+    local = (LOCAL_STORAGE *)storage_source;
+    match = NULL;
+
+    if (location_handle == NULL && name != NULL)
+        return local_err(local, session, EINVAL, "flush: cannot specify name without location");
+
+    local->op_count++;
+    if (location_handle != NULL) {
+        if ((ret = local_location_path(
+               local, location_handle, name == NULL ? "" : name, NULL, &match)) != 0)
+            goto err;
+    }
+    VERBOSE(local, "Flush: match=%s\n", SHOW_STRING(match));
+
+    /*
+     * Note: we retain the lock on the data structure while flushing all entries. This is fine for
+     * our local file implementation, when we don't have to do anything to flush, but for a cloud
+     * implementation, we'll want some way to not hold the lock while transferring data.
+     */
+    if ((ret = pthread_rwlock_wrlock(&local->flush_lock)) != 0) {
+        (void)local_err(local, session, ret, "flush: pthread_rwlock_wrlock");
+        goto err;
+    }
+
+    TAILQ_FOREACH_SAFE(flush, &local->flushq, q, safe_flush)
+    {
+        if (match != NULL) {
+            /* Make sure the location matches. */
+            if (name != NULL) {
+                /* Exact name match required. */
+                if (strcmp(flush->src_path, match) != 0)
+                    continue;
+            }
+            /* No name specified, everything up to the name must match. */
+            else if (strncmp(flush->src_path, match, strlen(match)) != 0)
+                continue;
+        }
+        if ((t_ret = local_flush_one(local, session, flush)) != 0 && ret == 0)
+            ret = t_ret;
+        local_flush_free(flush);
+        TAILQ_REMOVE(&local->flushq, flush, q);
+    }
+
+    if ((t_ret = pthread_rwlock_unlock(&local->flush_lock)) != 0) {
+        (void)local_err(local, session, t_ret, "flush: pthread_rwlock_unlock");
+        if (ret == 0)
+            ret = t_ret;
+    }
+
+err:
+    free(match);
+
+    return (ret);
+}
+
+/*
+ * local_flush_one --
+ *     Flush one item on the flush queue.
+ */
+static int
+local_flush_one(LOCAL_STORAGE *local, WT_SESSION *session, LOCAL_FLUSH_ITEM *flush)
+{
+    int ret;
+    char *object_name;
+
+    ret = 0;
+
+    object_name = strrchr(flush->src_path, '/');
+    if (object_name == NULL)
+        ret = local_err(local, session, errno, "%s: unexpected src path", flush->src_path);
+    else {
+        object_name++;
+
+        /* Here's where we would copy the file to a cloud object. */
+        VERBOSE(local, "Cloud flush: from=%s, bucket=%s, object=%s, kmsid=%s, \n", flush->src_path,
```

Comment:
While the comment is good, I think having "Cloud" in a local implementation output could be confusing. Maybe instead "Flush Object" and rename the structure field `object_flushes`? I won't push hard for this though.